### PR TITLE
icestorm: add 0.0

### PR DESCRIPTION
**Pull request by umarcor**
This PR adds package `icestorm`: https://github.com/YosysHQ/icestorm

### CODE REVIEW

**Comment by Biswa96** [comment]
Some comments:
1. Add a Maintainer comment at start.
2. If possible, a tarball can be used instead of git repository.
3. The `-j` option can be removed from `make` command. It is passed automatically with `$MAKEFLAGS`.

**Comment by umarcor** [comment]
> 1. Add a Maintainer comment at start.

Done.

> 2. If possible, a tarball can be used instead of git repository.

Unfortunately, `codeload.github.com` does not provide tarballs for untagged commits, and the "regular" tarballs are rate limited. That's why I think that using git as a source is "safer". 

> 3. The `-j` option can be removed from `make` command. It is passed automatically with `$MAKEFLAGS`.

Done.

File: mingw-w64-icestorm/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Maintainer: umarcor <unai.martinezcorral@ehu.eus>
+
+_realname=icestorm
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=da52117
```
Comment (lazka) [inline mingw-w64-icestorm/PKGBUILD:6]: That's not a proper version number. We need something that can be incremented over time so pacman knows when to update things.

File: mingw-w64-icestorm/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+         "${MINGW_PACKAGE_PREFIX}-libftdi")
+makedepends=("${MINGW_PACKAGE_PREFIX}-gcc" "flex")
+
+source=("icestorm::git://github.com/YosysHQ/${_realname}.git#commit=${pkgver}")
+sha256sums=('SKIP')
+
+build() {
+  cd "${srcdir}/${_realname}"
+  sed -i "s/-ggdb //;" Makefile
+  make STATIC=1
```
Comment (lazka) [inline mingw-w64-icestorm/PKGBUILD:22]: You need to set the build prefix to MINGW_PREFIX, otherwise this installs into /usr/local

**Comment by lazka** [comment]
It still installs to /usr/local...

**Comment by umarcor** [comment]
:sweat: Hope it's ok now...

**Comment by lazka** [comment]
It's now installing to /mingw64/usr/local instead of  /mingw64

**Comment by umarcor** [comment]
Please, insert huge facepalm here, and accept my apologies for the disturbances.

**Comment by lazka** [comment]
two small things:

* seems it installs various scripts using python, should python be a dependency?
* it installs a Python module to /bin instead of the standard place, I guess that's upstreams fault.. so nothing we can do about it..

**Comment by umarcor** [comment]
> * seems it installs various scripts using python, should python be a dependency?

icestorm includes several tools: fpga architecture documentation, bitstream packaging tools, RAM modification tools, FPGA/board programming tools... Python is only required when the FPGA architecture documentation resources are used, and that is done by a third party tool (nextpnr), which is the one that depends on Python. Therefore, I don't think it's necessary to make Python a dependency of icestorm itself.

> * it installs a Python module to /bin instead of the standard place, I guess that's upstreams fault.. so nothing we can do about it..

I believe it is to be considered a system tool. That is, the fact that it's written in Python is expected to be ignored by users. Anyway, I can ask about it upstream and fix/improve it in a separate PR, if you are ok with that.

**Comment by lazka** [comment]
ok. it contains 12 tools under /bin that depend on python.. if you are ok with those being broken..., your choice :)

**Pull request merged**